### DISCUSSION TITLE
How to test if an interrupt handler exist ?

### DISCUSSION THREAD

**Discussion by Tico06** [micropython/micropython#12798] [ESP32] [answered]
Hi There,

I tried to use disable_irq()/enable_irq() on ESP32 S2. Unfortunately, once disable_irq() is called, the backend reboots. I found in the Discussions it could be cause by no interrupt handler is available.

How could I test an interrupt handler exists ?

Thanks and regards,

Eric.

**Comment by jimmo** [comment, marked as answer]
It probably reboots because the watchdog fires. This means you're leaving interrupts disabled for too long.

**Reply by Tico06** [reply]
Hi Jimmo,

Thanks a lot for your answer.

What do you mean by too long ? It reboots right after calling disable_irq(), which doesn't happens with Raspberry PI Pico.

Thx and reg

Eric.

**Reply by jimmo** [reply]
Ideally no more than a few hundred microseconds.

i.e. you can't call disable_irq at the repl prompt.

What do you mean by "right after calling disable_irq" ?

**Reply by Tico06** [reply]
Well, I tested the following code:

print('Before disable_irq')
disable_irq()
print('After disable_irq')

The second print never had a chance to output something... But I guess with few hundred of microseconds, it's a challenge for it !

My intention is to read SPI, less than 100 bytes, at once. How could I proceed ?
